File: src/main/java/ru/sadv1r/afc/ideaPlugin/achievements/TenSymbolsTyped.java
package ru.sadv1r.afc.ideaPlugin.achievements;

import ru.sadv1r.afc.ideaPlugin.Achievable;

/**
 * Created on 12/10/15.
 *
 * @author sadv1r
 * @version 0.1
 */
public class TenSymbolsTyped implements Achievable {
    private int finish = 10;
    private final String name = "Find the keyboard";
    private final String text = "Type " + getFinish() + " symbols";

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getText() {
        return text;
    }

    @Override
    public int getProgress() {
        return stats.getSymbolsTyped() >= getFinish() ? finish : stats.getSymbolsTyped();
    }

    @Override
    public int getFinish() {
        return finish;
    }
}

File: src/main/kotlin/ru/sadv1r/afc/ideaPlugin/achievements/TenSymbolsTyped.kt
package ru.sadv1r.afc.ideaPlugin.achievements

import ru.sadv1r.afc.ideaPlugin.Achievable

class TenSymbolsTyped(
    name: String = "Find the keyboard",
    progress: Int = 0,
    finish: Int = 10,
    text: String = "Type $finish symbols"
): Achievable(name, text, progress, finish) {
    override val progress: Int
        get() = if (stats.symbolsTyped >= finish) finish else stats.symbolsTyped
}